testing/MLDB-2180-dataset-split.py

```diff
@@ -0,0 +1,184 @@
+#
+# MLDB-2180-dataset-split.py
+# Mathieu Marquis Bolduc, 2017-04-03
+# This file is part of MLDB. Copyright 2017 mldb.ai inc. All rights reserved.
+#
+
+mldb = mldb_wrapper.wrap(mldb)  # noqa
+
+class Mldb2180DatasetSplitTests(MldbUnitTest):  # noqa
+
+    @classmethod
+    def setUpClass(cls):
+        ds = mldb.create_dataset({'id' : 'ds1', 'type' : 'sparse.mutable'})
+        for i in xrange(4):
+            val = 'x' if i < 2 else 'y'
+            ds.record_row('%d' % i, [[val, 1, 0]])
+        ds.commit()
+
+        ds2 = mldb.create_dataset({'id' : 'ds2', 'type' : 'sparse.mutable'})
+        for i in xrange(20):
+            val = 'x' if i < 16 else 'y'
+            ds2.record_row('%d' % i, [[val, 1, 0]])
+        ds2.commit()
+
+        ds2 = mldb.create_dataset({'id' : 'ds3', 'type' : 'sparse.mutable'})
+        for i in xrange(20):
+            val = 'x' if i < 10 else 'y'
+            ds2.record_row('%d' % i, [[val, 1, 0]])
+        ds2.commit()
+
+        ds2 = mldb.create_dataset({'id' : 'ds4', 'type' : 'sparse.mutable'})
+        #10/24 x, 15/24 y, 9 / 24 z
+        for i in xrange(24):
+            if i < 5:
+                ds2.record_row('%d' % i, [['x', 1, 0]])
+            elif i < 8:
+                ds2.record_row('%d' % i, [['x', 1, 0], ['y', 1, 0]])
+            elif i < 10:
+                ds2.record_row('%d' % i, [['x', 1, 0], ['y', 1, 0], ['z', 1, 0]])
+            elif i < 17:
+                ds2.record_row('%d' % i, [['y', 1, 0]])
+            elif i < 20:
+                ds2.record_row('%d' % i, [['y', 1, 0], ['z', 1, 0]])
+            else:
+                ds2.record_row('%d' % i, [['z', 1, 0]])
+        ds2.commit()
+
+    #Test that we try to represent every label in every dataset, regardless of distribution
+
+    def test_spread(self):
+        mldb.put("/v1/procedures/split", {
+            "type": "split.train",
+            "params": {
+                "labels": "SELECT * FROM ds1",
```

Comment:
Since this will always be used with a select containing only a subset of columns as labels, I would add a test covering that.